packages/mui-material/src/Select/Select.test.js

```diff
@@ -79,6 +79,88 @@ describe('<Select />', () => {
     );
   });
 
+  it('should support conditional rendering with empty string', () => {
+    render(
+      <Select open value={2}>
+        {'' && <MenuItem value={1}>One</MenuItem>}
+        <MenuItem value={2}>Two</MenuItem>
+      </Select>,
+    );
+  });
+
+  it('should support conditional rendering with negative number', () => {
+    render(
+      <Select open value={2}>
+        {-1 && <MenuItem value={1}>One</MenuItem>}
```

Comment:
I don't think it's worth testing cases `-1` or `true`. React will simply see the MenuItem there.